boards/arm/udoo_neo_full_m4/doc/udoo_neo_full.rst

```diff
@@ -0,0 +1,216 @@
+  - 3-Axis Digital Gyroscope
+  - 1x Sensor Snap-In I2C connector
+
+- Expansion port
+
+  - Arduino interface
+
+For more information about the MCIMX6X SoC and UDOO Neo Full board,
+see these references:
+
+- `NXP i.MX 6SoloX Website`_
+- `NXP i.MX 6SoloX Datasheet`_
+- `NXP i.MX 6SoloX Reference Manual`_
+- `UDOO Neo Website`_
+- `UDOO Neo Getting Started`_
+- `UDOO Neo Documentation`_
+- `UDOO Neo Datasheet`_
+- `UDOO Neo Schematics`_
+
+Supported Features
+==================
+
+The UDOO Neo Full board configuration supports the following hardware
+features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| SYSTICK   | on-chip    | systick                             |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | general purpose input/output        |
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/udoo_neo_full_m4/udoo_neo_full_m4_defconfig``
+
+Other hardware features are not currently supported by the port.
+
+Connections and IOs
+===================
+
+The UDOO Neo Full board was tested with the following pinmux
+controller configuration.
+
++---------------+-----------------+---------------------------+
+| Board Name    | SoC Name        | Usage                     |
++===============+=================+===========================+
+| J4 RX         | UART5_RX_DATA   | UART Console              |
++---------------+-----------------+---------------------------+
+| J4 TX         | UART5_TX_DATA   | UART Console              |
++---------------+-----------------+---------------------------+
+
+System Clock
+============
+
+The MCIMX6X SoC is configured to use the 24 MHz external oscillator
+on the board with the on-chip PLL to generate core clock.
+PLL settings for M4 core are set via code running on A9 core.
+
+Serial Port
+===========
+
+The MCIMX6X SoC has six UARTs. UART5 is configured for M4 console and the
+remaining are used by A9 core or not used.
+
+Programming and Debugging
+*************************
+
+The M4 core does not have flash memory and is not provided a clock
+at power-on-reset. Therefore it needs to be started by the A9 core.
+The A9 core is responsible to load the M4 binary application into the RAM,
+put the M4 in reset, set the M4 Program Counter and Stack Pointer, and get
+the M4 out of reset. The A9 can perform these steps at bootloader level
+after the Linux system has booted.
```

Comment:
Need to add an "or".
`at the bootloader level or after the Linux system has booted.`